sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiClient } from '@mysten/sui.js/src/client';
+import { fetchKiosk, getOwnedKiosks } from '../query/kiosk';
+import {
+	type FetchKioskOptions,
+	KioskData,
+	OwnedKiosks,
+	Network,
+	type KioskClientOptions,
+} from '../types';
+import { TransferPolicyRule, PERSONAL_KIOSK_RULE_ADDRESS, rules } from '../constants';
+import { queryTransferPolicy } from '../query/transfer-policy';
+
+/**
+ * A Client that allows you to interact with kiosk.
+ * Offers utilities to query kiosk, craft transactions to edit your own kiosk,
+ * purchase, manage transfer policies, create new kiosks etc.
+ */
+export class KioskClient {
+	client: SuiClient;
+	network: Network;
+	rules: TransferPolicyRule[];
+
+	constructor(options: KioskClientOptions) {
+		this.client = options.client;
+		this.network = options.network;
+		this.rules = rules; // add all the default rules.
+	}
+
+	/// Querying
+
+	/**
+	 * Get an addresses's owned kiosks.
+	 * @param address The address for which we want to retrieve the kiosks.
+	 * @returns An Object containing all the `kioskOwnerCap` objects as well as the kioskIds.
+	 */
+	async getOwnedKiosks({ address }: { address: string }): Promise<OwnedKiosks> {
+		const personalPackageId = PERSONAL_KIOSK_RULE_ADDRESS[this.network];
```

Comment:
I've been thinking a bit about these constants, and I wonder if it would be better to set these in the client constructor, and allow them to be overwritten via options passed to the client.  This would allow the SDK to work in devnet/localnet as well (assuming you know how to lookup and pass in the proper ids)

Reply:
Agreed. I'll prepare an option to override the `packageIds` for each of the rules.